### PR TITLE
Created Swagger Doc for fd2_api; Refactored of cropmap and areamap in seeding report; Updated FarmData2API.js

### LINKED ISSUE #597: Express.js Cutsom API Endpoint Documentation
Documentation is needed for the Express.js Custom API endpoint. To allow JSDOC to automatically create documentation for the API Endpoints, we should consider including a doc folder inside the farmdata2_api folder. ~~Also, the README.md file inside the docker folder should be updated so that it includes information regarding the Express.js docker image and container.~~

### PR DESCRIPTION
**Pull request by JingyuMarcelLee**
1. Created Swagger documentation using swagger related Node modules. 
2. Created four API endpoints under app.js to use SQL queries to return JSON object for cropmap and areamap
3. Refactored FarmOSAPI.js and seedingreport to use the new API endpoints. 

__Pull Request Description__

Closes #597 #606 #607

---
__Licensing Certification__

FarmData2 is a [Free Cultural Work](https://freedomdefined.org/Definition) and all accepted contributions are licensed as described in the LICENSE.md file. This requires that the contributor holds the rights to do so. By submitting this pull request __I certify that I satisfy the terms of the [Developer Certificate of Origin](https://developercertificate.org/)__ for its contents.

### CODE REVIEW

**Comment by JingyuMarcelLee** [comment]
The docker image had to be rebuilt from the docker/api folder using the package.json from the farmdata2/farmdata2_api folder.

File: docker/api/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (braughtg) [inline docker/api/package.json:1]: It looks like this file is present twice.  So there is a question of where this file should live.  There should probably only be one copy of it and then we can move it to where it is needed (if different) during the build process.

Comment (JingyuMarcelLee) [reply docker/api/package.json]: Resolved by deleting the filed in the API container.

File: farmdata2/farmdata2_api/src/swagger.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      "url": "https://creativecommons.org/licenses/by-sa/4.0/"
+    },
+    "contact": {
+      "name": "FarmData2 GitHub Repository",
+      "url": "https://github.com/DickinsonCollege/FarmData2"
+    }
+  },
+  "servers": [
+    {
+      "url": "http://http://fd2_api/",
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/src/swagger.json:18]: fix

File: farmdata2/farmdata2_modules/resources/FarmOSAPI.js [hunk 3]
```diff
@@ -220,7 +241,14 @@ function getIDToAreaMap(){
...
-    return getMap('/taxonomy_term.json?bundle=farm_areas', 'name', 'tid')
+    return new Promise ((resolve, reject) => {
+        axios.get("http://fd2_api/areas/mapByName")
+        .then((response) => {
+            resolve(new Map(Object.entries(response.data)))
+        }).catch(function (err) {
+            reject(err)
+        })
+    })
 }
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_modules/resources/FarmOSAPI.js:253]: Use the existing unit test to test the functions

File: farmdata2/farmdata2_api/src/app.js [hunk 4]
```diff
@@ -20,8 +37,232 @@ app.listen(port, () => {
   console.log(`Server is up at port:${port}`);
 });
 
-// use : to specify route parameters
-app.get("/user/:userid?", async (req, res) => {
+/**
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/src/app.js:40]: test swagger-jsdoc comment for express (npm module)

File: farmdata2/farmdata2_api/src/app.js [hunk 5]
```diff
@@ -20,8 +37,232 @@ app.listen(port, () => {
...
+ *                           example: admin
+ *                         mail:
+ *                           type: string
+ *                           description: The email address of the user
+ *                           example: admin@example.com
+ */
+app.get("/users/:userid?", async (req, res) => {
   let conn;
   try {
     conn = await pool.getConnection();
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/src/app.js:268]: Do all the maps

File: farmdata2/farmdata2_api/src/app.js [hunk 6]
```diff
@@ -20,19 +32,134 @@ app.listen(port, () => {
...
-app.get("/user/:userid?", async (req, res) => {
+/**
+ * GET /crops/mapByName
+ * @summary Returns a mapping of crop names to their corresponding tid.
+ * @tags Maps
+ * @return {object} 200 - Returns a mapping of crop names to their corresponding tid.
+ * @example response - 200 - success response
+ * {
+ *  "BROCCOLI": "12"
+ * }
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/src/app.js:43]: More examples

File: farmdata2/farmdata2_api/src/app.js [hunk 7]
```diff
@@ -20,19 +32,134 @@ app.listen(port, () => {
...
+});
+
+/**
+ * GET /areas/mapById
+ * @summary Returns a mapping of area tid to their corresponding area names.
+ * @tags Maps
+ * @return {object} 200 - Returns a mapping of area tid to their corresponding area names.
+ * @example response - 200 - success response
+ * {
+ *  "12": "BROCCOLI"
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/src/app.js:145]: fix

File: farmdata2/farmdata2_api/cypress/areas.api.spec.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+describe("API Request Functions", () => {
+  beforeEach(() => {
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/cypress/areas.api.spec.js:2]: before

File: farmdata2/farmdata2_api/cypress/areas.api.spec.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    cy.request("/areas/mapbyName").as("mapByName");
+    cy.get("@mapByName").then((mapByName) => {
+      expect(mapByName.status).to.eq(200);
+      assert.isObject(mapByName.body, "mapByName Response is an object");
+      expect(Object.keys(mapByName.body)[0]).to.eq("A");
+      expect(
+        Object.keys(mapByName.body)[Object.keys(mapByName.body).length - 1]
+      ).to.eq("Z");
+      expect(Object.keys(mapByName.body).length).to.eq(70);
+      cy.log(JSON.stringify(mapByName.body));
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/cypress/areas.api.spec.js:17]: take logs out

File: farmdata2/farmdata2_api/cypress/crops.api.spec.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
+describe('API Request Functions', () => {
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_api/cypress/crops.api.spec.js:1]: make it more descriptive using collection names

File: farmdata2/farmdata2_modules/resources/FarmOSAPI.spec.js [hunk 11]
```diff
@@ -146,7 +146,7 @@ describe('API Request Functions', () => {
         })
     })
 
-    context('test maping functions', () => {
+    context.only('test maping functions', () => {
```
Comment (JingyuMarcelLee) [inline farmdata2/farmdata2_modules/resources/FarmOSAPI.spec.js:149]: fix

**Pull request closed without merging**